### PR TITLE
Cart v1

### PR DESCRIPTION
**Pull request by teresafinn**
First cut at implementing cart functionality

### CODE REVIEW

File: app/controllers/cart_items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class CartItemsController < ApplicationController
+
+  def create
+    item = Item.find(params[:item])
+    @cart.add_item(item)
+    session[:cart] = @cart.data
+    flash[:notice] = "You have #{pluralize(@cart.count_of(item), item.name)} in your cart."
```
Comment (mirjoy) [inline app/controllers/cart_items_controller.rb]: ignore hound!

File: app/models/cart.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class Cart
+  attr_reader :data
+
+  def initialize(input_data)
+    @data = input_data || Hash.new
+  end
+
+  def add_item(item)
+    data[item.id.to_s] ||= 0
```
Comment (indiesquidge) [inline app/models/cart.rb]: Pull this out into its own method

Comment (indiesquidge) [inline app/models/cart.rb]: Won't work replacing `data[item.id.to_s]` in this file. Need to ask about this.

File: db/seeds.rb [hunk 3]
```diff
@@ -1,7 +1,5 @@
-# This file should contain all the record creation needed to seed the database with its default values.
-# The data can then be loaded with the rake db:seed (or created alongside the db with db:setup).
-#
-# Examples:
-#
-#   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
-#   Mayor.create(name: 'Emanuel', city: cities.first)
+Item.create(name: "Fudge", description: "Delicious ooey gooey double dark chocolate brownie", price: 400)
+
+Item.create(name: "Mint Chocolate Chip Ice Cream", description: "Scrumptious mint chocolate chip ice cream with dark chocolate chips", price: 500)
```
Comment (indiesquidge) [inline db/seeds.rb]:
```ruby
Item.create(name: "Mint Chocolate Chip Ice Cream",
                    description: "Scrumptious mint chocolate chip ice cream with dark chocolate chips",
                    price: 500)
```

File: db/seeds.rb [hunk 4]
```diff
@@ -1,7 +1,5 @@
-# This file should contain all the record creation needed to seed the database with its default values.
-# The data can then be loaded with the rake db:seed (or created alongside the db with db:setup).
-#
-# Examples:
-#
-#   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
-#   Mayor.create(name: 'Emanuel', city: cities.first)
+Item.create(name: "Fudge", description: "Delicious ooey gooey double dark chocolate brownie", price: 400)
```
Comment (indiesquidge) [inline db/seeds.rb]:
```ruby
Item.create(name: "Fudge",
                   description: "Delicious ooey gooey double dark chocolate brownie",
                   price: 400)
```

File: db/seeds.rb [hunk 5]
```diff
@@ -1,7 +1,5 @@
...
+
+Item.create(name: "Mint Chocolate Chip Ice Cream", description: "Scrumptious mint chocolate chip ice cream with dark chocolate chips", price: 500)
+
+Item.create(name: "Moose Tracks", description: "Caramel and peanut butter chunks swirled in vanilla", price: 500)
```
Comment (indiesquidge) [inline db/seeds.rb]:
```ruby
Item.create(name: "Moose Tracks",
                   description: "Caramel and peanut butter chunks swirled in vanilla",
                   price: 500)
```

**Comment by indiesquidge** [comment]
Acked. Except we need to correct all the hound errors.

File: db/seeds.rb (hunk 4 above)
Comment (teresafinn) [inline db/seeds.rb]: Fixed, thank you!

**Pull request merged**